CCMSPUI-376: Migrate notification summary from SOA-API to EBS-API

JamieBriggs-MoJ opened the pull request:
As part of CBP-947, work was done to move the notification summary from SOA to EBS, which resulted in a new view XXCCMS_NOTIFICATION_COUNT_V being created to support this. 

As part of this PR and CCMSPUI-376, the old method in SoaApiClient has been removed, and a new method has been added in EbsApiClient to use the new endpoint added as part of this PR:
https://github.com/ministryofjustice/laa-ccms-data-api/pull/119

JamieBriggs-MoJ commented:
The class `NotificationSummary` is still being used from the SOA gateway API rather than the new one generated in the EBS API so leaving this PR as a draft. It would for myself to first understand what the process for releasing new versions of APIs is before updating the versions within this project's `build.gradle`.

farrell-m reviewed (approved):
Approved (subject to the `NotificationsSummary` model being updated)

PhilDigitalJustice commented:
Are you able to amend the gradle build with the snapshot versions:

uk.gov.laa.ccms.data:data-api:0.0.21-9a44d8f-SNAPSHOT
uk.gov.laa.ccms.soa.gateway:soa-gateway-api:0.0.38-cfdf45d-SNAPSHOT

Usually do this, then once this is merged, a PR is auto raised for the release version for ebs-api and soa-api.

PhilDigitalJustice reviewed (changes requested):
Just the 1 change to check first

JamieBriggs-MoJ commented:
> Are you able to amend the gradle build with the snapshot versions:
> 
> uk.gov.laa.ccms.data:data-api:0.0.21-9a44d8f-SNAPSHOT uk.gov.laa.ccms.soa.gateway:soa-gateway-api:0.0.38-cfdf45d-SNAPSHOT
> 
> Usually do this, then once this is merged, a PR is auto raised for the release version for ebs-api and soa-api.

@PhilDigitalJustice I have updated build.gradle properties to the published SNAPSHOT versions and updated references to NotificationSummary to the new EBS varient.

JamieBriggs-MoJ commented:
After requesting review, realised the integration tests were not working due to old Soa method not existing anymore. Took the time to remove this, and also add the EBS integration tests for the new NotificationSummary endpoint as this was missed off initially.

PhilDigitalJustice reviewed (approved):
LGTM
